File: ForestManagement/App.DTO/WorkOrder/WorkOrderDto.cs
using App.Contracts.Enums;

namespace App.DTO.WorkOrder;

public class WorkOrderDto
{
    public Guid Id { get; set; }

    public Guid AssignedToId { get; set; }
    public string AssignedToUserName { get; set; } = string.Empty;

    public Guid AssignedById { get; set; }
    public string AssignedByUserName { get; set; } = string.Empty;

    public Guid ActivityTypeId { get; set; }
    public string ActivityTypeName { get; set; } = string.Empty;

    public Guid? ForestStandId { get; set; }
    public int? ForestStandNumber { get; set; }

    public Guid CadasterId { get; set; }
    public string CadasterCadastralNumber { get; set; } = string.Empty;

    public EOrderStatus Status { get; set; }

    public decimal Quantity { get; set; }

    public string? Unit { get; set; }

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: frontend/src/lib/dtos/workorder/workorder.dto.ts
export type WorkOrderStatus = 'Sent' | 'Completed';

export type WorkOrderCreateDto = {
	assignedToId: string;
	activityTypeId: string;
	cadasterId: string;
	forestStandId: string | null;
	quantity: number;
	unit: string | null;
	notes: string | null;
};

export type WorkOrderUpdateDto = {
	id: string;
	assignedToId: string;
	activityTypeId: string;
	cadasterId: string;
	forestStandId: string | null;
	quantity: number;
	unit: string | null;
	notes: string | null;
};

export type WorkOrderDto = {
	id: string;
	assignedToId: string;
	assignedToUserName: string;
	assignedToUserFirstName: string | null;
	assignedToUserLastName: string | null;
	assignedById: string;
	assignedByUserName: string;
	assignedByUserFirstName: string | null;
	assignedByUserLastName: string | null;
	activityTypeId: string;
	activityTypeName: string;
	forestStandId: string | null;
	forestStandNumber: number | null;
	cadasterId: string;
	cadasterCadastralNumber: string;
	status: WorkOrderStatus;
	quantity: number;
	unit: string | null;
	notes: string | null;
	createdAt: string;
};

export type WorkOrderListDto = {
	id: string;
	assignedToUserName: string;
	assignedToUserFirstName: string | null;
	assignedToUserLastName: string | null;
	activityTypeName: string;
	cadasterCadastralNumber: string;
	cadasterId: string;
	forestStandNumber: number | null;
	forestStandId: string | null;
	status: WorkOrderStatus;
	quantity: number;
	unit: string | null;
	createdAt: string;
};
